### PR TITLE
Preload; resolves #963

### LINKED ISSUE #963: Asset preloading before play

### PR DESCRIPTION
**Pull request by kschzt**

### CODE REVIEW

File: browser/plugins/stereo_cube_map.plugin.js [hunk 1]
```diff
@@ -123,10 +121,6 @@
 			that.rightObj.channels.set(that.rightChannel)
 
 			that.updated = true
-		},
-		undefined,
-		function(e) {
```
Comment (kkananen) [inline browser/plugins/stereo_cube_map.plugin.js]: what happens on errors now? can we hook up an error handler?

File: browser/scripts/loaders/loaders.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	if (!xhr.total) {
+		lengthKnown = false
+		return;
+	}
+
+	this.emit('progress', xhr.loaded / xhr.total)
+}
+
+// ----------------------------------------
+function ImageLoader(url) {
```
Comment (kkananen) [inline browser/scripts/loaders/loaders.js]: maybe *Loader could be split into their own files

File: browser/plugins/stereo_cube_map.plugin.js (hunk 1 above)
Comment (kschzt) [inline browser/plugins/stereo_cube_map.plugin.js]: yeah, I'll add a console.error() in AssetLoader

File: browser/scripts/loaders/assetLoader.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	// 'url_json_generator': 'json',
+	// 'url_video_generator': 'video'
+}
+
+// ------------------------------------------------
+
+function AssetLoader(loaders) {
+	EventEmitter.call(this)
+
+	this.loadingTexture = THREE.ImageUtils.loadTexture('/data/textures/loadingtex.png')
```
Comment (kkananen) [inline browser/scripts/loaders/assetLoader.js]: please do:
	this.defaultTexture.wrapS = THREE.RepeatWrapping
	this.defaultTexture.wrapT = THREE.RepeatWrapping

and the same for loadingTexture to enable repeat wrapping

**Comment by kkananen** [comment]
lgtm; please fix the wrapping though

**Pull request merged**